### PR TITLE
Proposal to fix the issue #190 

### PR DESCRIPTION
**Pull request by rafael-matsuda**
Proposal to fix the issue #190 using the same native Android SDK implementation. Please, let's discuss the best solution for this issue.

### CODE REVIEW

**Comment by Jeasmine** [comment]
Hey! Thank you for your contribution! I agree with your fix for the current problem. I have made this change https://github.com/OneSignal/OneSignal-Flutter-SDK/pull/259 to fix the failing testing. Let's wait for it to be merge and then after your tests are passing. Can you please squash and add a description to the commit in that way in the future we know what is this fixing? Thanks!

**Comment by rafael-matsuda** [comment]
> Hey! Thank you for your contribution! I agree with your fix for the current problem. I have made this change #259 to fix the failing testing. Let's wait for it to be merge and then after your tests are passing. Can you please squash and add a description to the commit in that way in the future we know what is this fixing? Thanks!

Hi Jeasmine! Thank you for you reply! It's a pleasure to help your team. Sure, I'll wait you PR to be merged and then I will squash and add the description. Thank you!

File: android/src/main/java/com/onesignal/flutter/FlutterRegistrarResponder.java [hunk 1]
```diff
@@ -1,7 +1,11 @@
 package com.onesignal.flutter;
 
+import android.os.Handler;
+import android.os.Looper;
+
 import android.app.Activity;
 
+
```
Comment (mikechoch) [inline android/src/main/java/com/onesignal/flutter/FlutterRegistrarResponder.java:8]: Get rid of extra new line

File: android/src/main/java/com/onesignal/flutter/FlutterRegistrarResponder.java [hunk 2]
```diff
@@ -54,7 +58,12 @@ public void run() {
    }
 
    private void runOnMainThread(final Runnable runnable) {
-      ((Activity) flutterRegistrar.activeContext()).runOnUiThread(runnable);
+      if (Looper.getMainLooper().getThread() == Thread.currentThread())
+         runnable.run();
+      else {
+         Handler handler = new Handler(Looper.getMainLooper());
+         handler.post(runnable);
+      }
```
Comment (mikechoch) [inline android/src/main/java/com/onesignal/flutter/FlutterRegistrarResponder.java:66]: Looks good, as discussed in #190 this `runOnMainThread` method now matches the exact native Android SDK implementation

**Comment by Jeasmine** [comment]
Test fix was rebased to master, can you please rebase? After that, can you squash your commits to only one with the description of the fix? Thanks!

**Pull request merged**